### PR TITLE
feat(firestore): Add Firestore Multi Database Support

### PR DESCRIPTION
**Pull request by jonathanedey**
Fixes: #718

### CODE REVIEW

File: firebase_admin/firestore.py [hunk 1]
```diff
@@ -18,59 +18,74 @@
...
     Raises:
-      ValueError: If a project ID is not specified either via options, credentials or
-          environment variables, or if the specified project ID is not a valid string.
+      ValueError: If the database ID is not None or a string or a project ID is not specified
+          either via options, credentials or environment variables, or if the specified
+          project ID is not a valid string.
 
-    .. _Firestore Client: https://googlecloudplatform.github.io/google-cloud-python/latest\
-          /firestore/client.html
+    .. _Firestore Client: https://cloud.google.com/python/docs/reference/firestore/latest/\
```
Comment (jonathanedey) [inline firebase_admin/firestore.py:57]: @egilmorez, Could you point me in the direction of the correct reference links for these. Think these should be it but not sure.

**Review by lahirumaramba — APPROVED** [review]
LGTM! Thanks!

File: firebase_admin/firestore.py (hunk 1 above)
Comment (egilmorez) [reply firebase_admin/firestore.py:57]: Ooh . . . honestly, I can't.  But Mark Arndt or Juan Lara on the Cloud writing team would probably know the lay of that land.

File: firebase_admin/firestore_async.py [hunk 2]
```diff
@@ -18,65 +18,74 @@
...
 
-def client(app: App = None) -> firestore.AsyncClient:
+def client(app: Optional[App] = None, database_id: Optional[str] = None) -> firestore.AsyncClient:
     """Returns an async client that can be used to interact with Google Cloud Firestore.
 
     Args:
-      app: An App instance (optional).
+        app: An App instance (optional).
+        database_id: The database ID of the Google Cloud Firestore database to be used.
+            Defaults to the default firestore database if not specified or empty string (optional).
```
Comment (egilmorez) [inline firebase_admin/firestore_async.py:47]: Here's the same caps issue Lahiru found in firestore.py.

**Review by egilmorez — APPROVED** [review]
LGTM!  But for a real answer on Python reference links, reach out to Mark or Juan Lara, thanks!

File: firebase_admin/firestore.py (hunk 1 above)
Comment (markarndt) [reply firebase_admin/firestore.py:58]: This is correct.

**Pull request merged**